File: Examples/test-suite/perl5/smart_pointer_rename_runme.pl
use strict;
use warnings;
use Test::More tests => 5;
BEGIN { use_ok('smart_pointer_rename') }
require_ok('smart_pointer_rename');

my $f = smart_pointer_rename::Foo->new();
my $b = smart_pointer_rename::Bar->new($f);

is($b->test(), 3);
is($b->ftest1(1), 1);
is($b->ftest2(2, 3), 2);

File: Examples/test-suite/lua/smart_pointer_rename_runme.lua
require("import")
require("smart_pointer_rename")
spr=smart_pointer_rename
catch_undef_globs() -- catch "undefined" global variables

foo = spr.Foo()
assert(foo:ftest1(1) == 1)
assert(foo:ftest2(1,2) == 2)

bar = spr.Bar(foo)
assert(bar:test() == 3)
assert(bar:ftest1(1) == 1)
assert(bar:ftest2(1,2) == 2)
